### PR TITLE
Integrate dpdk runmode v9

### PR DESCRIPTION
**Pull request by lukashino**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4507) ticket:

Describe changes:
- Integrate DPDK capture interface as an alternative to AF_PACKET or others.
- Focus implementation on workers run mode.
- Support both IDS and IPS modes.
- Follow AF_PACKET capture interface architecture and apply DPDK best practices for development.
- Update documentation to reflect new changes.
- Edit Github workflow to include build for the DPDK capture interface.

Work relies on pkg-config and requires at least DPDK 19.11. Work is focused on adding support for physical NICs. At the same time, it should be run natively (and not in docker or in virtual machines) as a primary process. To support multiple workers, the NICs need to support RSS offload.

Changelog from the previous [PR](https://github.com/OISF/suricata/pull/6611):
- created new PMD specific util files that - for now - contains setting of RSS setting on different Intel NICs. Mellanox does not require such files so far. 
- refactoring and responding to Victor's comments
- cherry-picked Victor's comments for easier debugging when RSS hash functions are not equal
- updated counters

### CODE REVIEW

**Comment by lukashino** [comment]
Hi @victorjulien, build on Cocci is failing. I am unsure why but to me it seems it has appeared in some other PRs as well. Do you know why?

**Comment by victorjulien** [comment]
> Hi @victorjulien, build on Cocci is failing. I am unsure why but to me it seems it has appeared in some other PRs as well. Do you know why?

We're looking into that. You can ignore it.

File: src/source-dpdk.c [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+    if (ptv->queue_id == 0) {
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_rx_errs,
+                eth_stats.imissed + eth_stats.ierrors + eth_stats.rx_nombuf + ptv->pkts);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_imissed, eth_stats.imissed);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_rx_no_mbufs, eth_stats.rx_nombuf);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_ierrors, eth_stats.ierrors);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_tx_errs, eth_stats.oerrors);
+        SC_ATOMIC_SET(ptv->livedev->drop, eth_stats.imissed + eth_stats.ierrors +
+                                                  eth_stats.rx_nombuf + eth_stats.oerrors +
+                                                  ptv->pkts);
```
Comment (amaan-fmad) [inline src/source-dpdk.c:290]: Hey! Seems like we're adding `ptv->pkts` to the `ptv->livedev->drop` and the `ptv->capture_dpdk_rx_errs` mistakenly?
AFAICT `ptv->pkts` is just the count of packets received from `rte_eth_rx_burst` (in `ReceiveDPDKLoop`), and shouldn't be included in error/drop counts?

**Comment by victorjulien** [comment]
Merged as part of #6708, thanks a lot @lukashino!

**Pull request closed without merging**